client/core/trade.go

```diff
@@ -373,6 +375,16 @@ func (t *trackedTrade) negotiate(msgMatches []*msgjson.Match) error {
 		// Set the order status for both orders.
 		t.metaData.Status = order.OrderStatusCanceled
 		t.db.UpdateOrderStatus(t.cancel.ID(), order.OrderStatusExecuted)
+		// If the order's backing coins are unused, unlock them, otherwise they
+		// have been or will be spent by a swap.
+		if !includesTrades && initFilled == 0 {
+			for _, coin := range t.coins {
+				err = t.wallets.fromWallet.ReturnCoins(asset.Coins{coin})
+				if err != nil {
+					log.Warnf("unable to unlock coin %v: %v", coin, err)
+				}
+			}
+		}
```

Comment:
This also has nothing to do with the main goals of the PR, but I notices that ReturnCoins wasn't being used anywhere it was designed for.

Reply:
This isn't the first time this has come up recently. I'll take care of it.

Reply:
Would you like to handle this and the change coin locking that is started in https://github.com/decred/dcrdex/pull/511?  I can remove these lines from `(*trackedTrade).negotiate`, but keeping the `ReturnCoins` call in `handleRevokeMatchMsg`, and close 511 (or you can cherry pick the commit and build on it, your call).

Reply:
That works.

Reply:
Done.  Just commit number 98cc3469844e91966bd009841c43e30682e46a07 is potentially of interest for that.